File: VB/GridControlWithBar/PropertyGridControlDescendant/VGridPainterDescendant.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraVerticalGrid.Painters
Imports DevExpress.XtraVerticalGrid.ViewInfo

Namespace GridControlWithBar
	Public Class VGridPainterDescendant
		Inherits VGridPainter
		Public Sub New(ByVal eventHelper As DevExpress.XtraVerticalGrid.Painters.PaintEventHelper)
			MyBase.New(eventHelper)

		End Sub

		Protected Overrides Sub DoDrawCore(ByVal vi As DevExpress.XtraVerticalGrid.ViewInfo.BaseViewInfo)
			MyBase.DoDrawCore(vi)
			DrawButton(vi)

		End Sub
		Protected Sub DrawButton(ByVal vi As BaseViewInfo)
			If (CType(vi.Grid, PropertyGridControlDescendant)).ButtonInfoList IsNot Nothing Then
			For Each info As EditorButtonObjectInfoArgs In (CType(vi.Grid, PropertyGridControlDescendant)).ButtonInfoList
				If info IsNot Nothing Then
					Dim painter As EditorButtonPainter = EditorButtonHelper.GetPainter(BorderStyles.Default)
					info.Cache = Me.DrawInfo.Cache
					painter.DrawObject(info)
				End If
			Next info
			End If
		End Sub


	End Class
End Namespace

File: CS/GridControlWithBar/PropertyGridControlDescendant/VGridPainterDescendant.cs
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraVerticalGrid.Painters;
using DevExpress.XtraVerticalGrid.ViewInfo;

namespace GridControlWithBar
{
    public class VGridPainterDescendant : VGridPainter
    {
        public VGridPainterDescendant(DevExpress.XtraVerticalGrid.Painters.PaintEventHelper eventHelper)
            : base(eventHelper)
        {
           
        }

        protected override void DoDrawCore(DevExpress.XtraVerticalGrid.ViewInfo.BaseViewInfo vi)
        {
            base.DoDrawCore(vi);
            DrawButton(vi);

        }
        protected  void DrawButton(BaseViewInfo vi)
        {
            if (((PropertyGridControlDescendant)vi.Grid).ButtonInfoList != null)
            foreach (EditorButtonObjectInfoArgs info in ((PropertyGridControlDescendant)vi.Grid).ButtonInfoList)
            {
                if (info != null)
                {
                    EditorButtonPainter painter = EditorButtonHelper.GetPainter(BorderStyles.Default);
                    info.Cache = this.DrawInfo.Cache;
                    painter.DrawObject(info);
                }
            }
        }

       
    }
}
